doc/charm++/manual.rst

```diff
@@ -8785,6 +8785,198 @@ following example shows how this API can be used.
+accelerator devices controlled by code on the CPU. Traditionally,
+programmers have had to either (a) halt the execution of work on the CPU
+whenever issuing GPU work to simplify synchronization or (b) issue GPU
+work asynchronously and carefully manage and synchronize concurrent GPU
+work in order to ensure progress and good performance. The latter becomes
+significantly more difficult with overdecomposition as in Charm++, where
+numerous concurrent objects launch computational kernels and initiate
+data transfers on the GPU.
+
+The support for GPUs in the Charm++ runtime system consist of the
+**GPU Manager** module and **Hybrid API (HAPI)**. Currently only NVIDIA GPUs
+(and CUDA) are supported, although we are actively working on providing
+support for AMD and Intel GPUs as well. CUDA code can be integrated in
+Charm++ just like any C/C++ program to offload computational kernels,
+but when used naively, performance will most likely be far from ideal.
+This is because overdecomposition, a core concept of Charm++, creates
+relatively fine-grained objects and tasks that needs support from the
+runtime to be executed efficiently.
+
+We strongly recommend using CUDA Streams to assign one more streams
+to each chare, so that multiple GPU operations initiated by different
+chares can be executed concurrently when possible. Concurrent Kernel
+Execution in CUDA allows kernels in different streams to execute
+concurrently on the device unless a single kernel uses all of the available
+GPU resources. It should be noted that concurrent data transfers (even when
+they are in different streams) are limited by the number of DMA engines
+on the GPU, and most current GPUs have only one engine per direction
+(host-to-device, device-to-host).
+
+In addition to using CUDA Streams to maximize concurrency, another
+important consideration is avoiding synchronization calls such as
+``cudaStreamSynchronize`` or ``cudaDeviceSynchronize``. This is because
+the chare that has just enqueued some work to the GPU should yield the
+PE, in order to allow other chares waiting on the same PE to execute.
+Because of the message-driven execution pattern in Charm++, it is infeasible
+for the user to asynchronously detect when a GPU operation completes,
+either by manually polling the GPU or adding a CUDA Callback to the stream.
+One of the core functionalities of GPU Manager is supporting this asynchronous
+detection, which allows a Charm++ callback to be invoked when all previous
+operations in a specified CUDA stream complete. This is exposed to the
+user via a HAPI call, which is demonstrated in the usage section below.
+
+Enabling GPU Support
+~~~~~~~~~~~~~~~~~~~~
+
+GPU support via GPU Manager and HAPI is not included by default when
+building Charm++. Use ``buildold`` with the ``cuda`` option to build Charm++
+with GPU support (CMake build is currently not supported), e.g.
+
+.. code-block:: bash
+
+   $ ./buildold charm++ netlrts-linux-x86_64 cuda -j8 --with-production
```

Comment:
I feel like mentioning the need for buildold in the manual is an anti-pattern. See #3009 which avoids the need for CUDA users to handle this extra detail.

Otherwise, LGTM.